### DISCUSSION TITLE
Add configuration option for custom transaction storage

### DISCUSSION THREAD

**Discussion by mfrey43** [auth0/auth0-spa-js#1424] [General]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/auth0-spa-js#readme), [Examples](https://github.com/auth0/auth0-spa-js/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-spa-js/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://auth0.com/docs/libraries/auth0-single-page-app-sdk) and [API documentation](https://auth0.github.io/auth0-spa-js/), and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-spa-js/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

My Ionic app uses the Capacitor Browser extension as described in this official guide: https://auth0.com/docs/quickstart/native/ionic-angular/01-login

Sometimes after the login flow completes and the inAppBrowser closes, the callback handling fails with the error 'missing_transaction'. This is because the Android system has killed my app to free up memory while it was in the background. When the callback reopens the app the transaction data is lost because it was saved in SessionStorage. My problem is that I can't change the storage location to something that isn't wiped during a restart. I can only configure cookie or session storage.

### Describe the ideal solution

I would like to set my own transactionStorage in the AuthorizationParams config object similar to how the cache configuration works. I already created a proof of concept in this commit: https://github.com/mfrey43/auth0-spa-js/commit/90da31095ed4c6d21166dd8d8956cffdd724862d

Alternatively, if I had the option to configure the transaction manager to use the cache for storage that would also work for me. I already configured the cache to use the device's native encrypted storage option.

### Additional context

I use this library directly without the auth0-angular wrapper because I needed more control.

**Comment by gyaneshgouraw-okta** [comment]
Hi @mfrey43,

Thank you for this well-detailed feature request and the proof-of-concept implementation! 

We appreciate the thorough research you've done and the clear problem statement, highlighting `missing_transaction` error in mobile environments where the OS kills background apps.

This is a pretty specific case (mobile hybrid apps with strict memory limits), so we'd like to move this to GitHub Discussions first. This will help us see if others face the same issue, gather more use cases, and get feedback on the proposed API.

Your proof-of-concept looks fine, so this feature seems doable if we move forward.
Thanks again for the detailed proposal! 🙏
